app/routes/goal_routes.py

```diff
@@ -1 +1,93 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response, abort, make_response
+from app.models.goal import Goal
+from app.models.task import Task
+from ..db import db
+from .route_utilities import validate_model
+
+bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@bp.post("")
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+    
+    return {"goal": new_goal.to_dict()}, 201
+
+@bp.get("")
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = [goal.to_dict() for goal in goals]
+    return goals_response
+
+@bp.get("/<goal_id>")
+def get_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal": goal.to_dict()}, 200
+
+@bp.put("/<goal_id>")
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
+
+@bp.delete("/<goal_id>")
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    db.session.delete(goal)
+    db.session.commit()
+
+    return Response(status=204, mimetype="application/json")
+
+
+
+@bp.route("/<goal_id>/tasks", methods=["POST"])
```

Comment:
🧐 You're mixing Flask syntax.

We prefer to use `.post()`, like you did on line 9 in this file, which is a shorthand for using `.route()` and passing in `methods` (see docs [here](https://flask.palletsprojects.com/en/stable/api/#flask.Flask.post)). We should be sure to not introduce inconsistencies in our codebase which it make it harder to maintain and unpredictable. 

Take care to use the syntax that a project already has OR follow the most up to date syntax per the library's documentation.

```suggestion
@goals_bp.post("/<goal_id>/tasks")
```